Create Obs form phase 1: images and EXIF data

nimmolo opened the pull request:
Revamp of the obs edit/create form image section.

Part 1 of 3.

nimmolo committed:
Reorder images in form

nimmolo committed:
start outline rearrangement

nimmolo committed:
date_select_with_label

nimmolo committed:
Change img_messages to date_messages, simplify images section

nimmolo committed:
Update _good_image_fields.html.erb

nimmolo committed:
Add observations/images/uploads/new and show

nimmolo committed:
Update _template.erb

nimmolo committed:
Update _good_image_fields.html.erb

nimmolo committed:
Fix controller on date_select

nimmolo committed:
Update _good_image_fields.html.erb

nimmolo committed:
Merge branch 'nimmo-date-select-label' into create-obs-reorg

nimmolo committed:
Merge branch 'nimmo-date-select-label' into create-obs-reorg

nimmolo committed:
Update forms_helper.rb

nimmolo committed:
Update _good_image_fields.html.erb

nimmolo committed:
Merge branch 'main' into create-obs-reorg

nimmolo committed:
Merge branch 'nimmo-send-file-info-to-template' into create-obs-reorg

nimmolo committed:
remove inline style duplicate

nimmolo committed:
Merge branch 'nimmo-refactor-carousels' into create-obs-reorg

nimmolo committed:
Create _edit_carousel.erb

nimmolo committed:
Merge branch 'nimmo-refactor-carousels' into create-obs-reorg

nimmolo committed:
Update _edit_carousel.erb

nimmolo committed:
work on carousel_with_forms

nimmolo committed:
Merge branch 'nimmo-refactor-carousels' into create-obs-reorg

nimmolo committed:
try

nimmolo committed:
improve

nimmolo committed:
more local

nimmolo committed:
more tries

nimmolo committed:
better comments in js

nimmolo committed:
render Template as turbo stream, alter controller, css

nimmolo committed:
refine

nimmolo committed:
controls position

nimmolo committed:
Merge branch 'main' into create-obs-reorg

nimmolo committed:
sortCarousel

nimmolo committed:
Update obs-form-images_controller.js

indicators
targetDisconnected

nimmolo committed:
Refactor set_thumbnail UX, get ready for syncing EXIF to obs

nimmolo committed:
Get good images ready

nimmolo committed:
Merge branch 'nimmo-try-single-image-detach-route' into create-obs-reorg

nimmolo committed:
Detach on edit

clearer CSS

nimmolo committed:
streamline templates

nimmolo committed:
Transfer exif to obs working

nimmolo committed:
Use exif from first loaded image

nimmolo committed:
Update observation.rb

Add Observation instance method `images_sorted`

nimmolo committed:
Update edit_and_update.rb

Use images_sorted

nimmolo committed:
Update _images.erb

Use images_sorted

nimmolo committed:
Update show.html.erb

Use images_sorted for carousel

nimmolo committed:
Update show.rb

Move @images to controller action

nimmolo committed:
Update show.html.erb

Use @images for carousel

nimmolo committed:
Update _carousel.erb

Remove top_img arg

nimmolo committed:
Update _carousel.erb

Remove top_img arg

nimmolo committed:
Update _carousel.erb

One more

nimmolo committed:
Update _confident_observations_panel.html.erb

Remove top_img arg

nimmolo committed:
Update _best_images.erb

rm top_img arg

nimmolo committed:
Update matrix_box_helper.rb

rm top_img arg

nimmolo committed:
Update obs-form-images_controller.js

nimmolo committed:
Merge branch 'create-obs-reorg' of https://github.com/MushroomObserver/mushroom-observer into create-obs-reorg

nimmolo committed:
Update _images.erb

use @images in soon-to-be-retired panel

nimmolo committed:
Fix the function to transfer the exif data to the obs

nimmolo committed:
Merge branch 'image-exif-data' into create-obs-reorg

nimmolo committed:
Add image exif info to obs edit screen

nimmolo committed:
Call exif info directly from item

nimmolo committed:
Merge branch 'main' into create-obs-reorg

nimmolo committed:
Merge branch 'main' into create-obs-reorg

nimmolo committed:
remove radios

nimmolo committed:
Update _images.html.erb

clean up

nimmolo committed:
Update _images.html.erb

Simplify

nimmolo committed:
Update _images.html.erb

nimmolo committed:
make something happen on exif copy

nimmolo committed:
Refactor for thumbnails tray

nimmolo committed:
Merge branch 'main' into create-obs-reorg

nimmolo committed:
Panel for images

nimmolo committed:
split controllers

nimmolo committed:
Refine carousel css

nimmolo committed:
edit form cleanup

set dataset.geocode in edit form item, as for new item
add "procesed" event
pre-round the exif lat/lng to MO precision, even out float rounds in app everywhere (to 4)

nimmolo committed:
Css for empty carousels

nimmolo committed:
Update lightbox_helper.rb

Fixes for no lightbox data (uploaded image)

nimmolo committed:
Update edit_and_update.rb

Fix for no good images

nimmolo committed:
Update _carousel_item.erb

Fix data-action syntax

nimmolo committed:
Update observations_controller_test.rb

Pass a good_images param

nimmolo committed:
Fixes to some CSS and layout issues i hadn't noticed

mo-nathan commented:
Looking at the latest on this branch, I find the initial layout of the “Select file…” button to be confusing and generally not great.  If I haven’t selected any images, I’d rather just see a “Select Images” button and not an empty panel with the title “Images” and a button way over on the right hand side of the screen.

nimmolo committed:
Add good_images param to tests of project_checks

nimmolo committed:
Merge branch 'main' into create-obs-reorg

nimmolo committed:
Update _images.html.erb

nimmolo committed:
Update _images.html.erb

rewrite without concats, and add comments

nimmolo committed:
Merge branch 'main' into create-obs-reorg

nimmolo committed:
Comment out radio button tests

nimmolo committed:
Obs form: Put field code in p tag

nimmolo committed:
System test: file upload

nimmolo committed:
Rename remove_image_button, use link_icon helper

nimmolo committed:
Update form-exif_controller.js

Round lat/lng to 4

nimmolo committed:
Update application_system_test_case.rb

Disable css animations in Capybara

nimmolo committed:
Ugh try to fix tests

nimmolo committed:
attempts

nimmolo committed:
Fix thumbnail order and "thumb_image_id" radio button field names

nimmolo committed:
Add some translations

nimmolo committed:
scss file field

nimmolo committed:
Update uploads_controller.rb

remove commented-out lines

nimmolo committed:
Date select - guard against year 0000

nimmolo committed:
`get_exif_data` in the controller

send `images` to carousel partial
send `image` as param to carousel_item partial
add class `carousel-item` to item
form-exif controller: rename event "processed" to "populated"
simplify `click_attach_file`
pass `thumb_id` to carousel_set_thumb_img_button for `checked`
remove_image_button use named param
test: there's now an earlier flash for images being uploaded, when obs commit doesn't pass validation

nimmolo committed:
Update observation_form_system_test.rb

Remove old comment

nimmolo committed:
exif data backup values empty hash

edit-carousel -> image-form-carousel

nimmolo committed:
param and arg name changes

param good_images -> good_image_ids
data-good-image -> data-image-status = "good"
data-added-image -> data-image-status = "upload"
carousel_item local exif_data -> camera_info: { file_name:, file_size }
date_select_with_label: use wrapping div as identifier

nimmolo committed:
Cleanup partials, ivars

nimmolo committed:
Update account_integration_test.rb

Fix odd error not finding select option

nimmolo committed:
rubocop cleanup, unused form-image targets

nimmolo committed:
class, target, param simplifications

nimmolo committed:
Merge branch 'main' into create-obs-reorg

nimmolo committed:
Update BlackOnWhite.scss

nimmolo committed:
form-exif-controller guard against NaN lat/lng/alt

nimmolo committed:
Update form-exif_controller.js

nimmolo committed:
exif controller, Image.read_exif_geocode, camera_info template

lat, lng, alt: clean up NaN and decimal places, move "m" to the template

nimmolo commented on app/models/image.rb:
```diff
@@ -782,23 +783,52 @@ def strip_gps!
+
+    { lat:, lng:, alt:, date:, file_name: nil, file_size: }
+  end
+
```
TODO: Write a unit test for this new method, using the `geocoded.jpg` file from our repo.

mo-nathan commented:
Hmm, I tried running it and the large images aren't showing up.  The smaller images along the bottom are (when there's more than one of them).
<img width="1183" alt="Screenshot 2024-06-14 at 7 49 03 PM" src="https://github.com/MushroomObserver/mushroom-observer/assets/3211730/82511414-5674-4204-9567-4f84d9aa0adc">
<img width="723" alt="Screenshot 2024-06-14 at 7 47 56 PM" src="https://github.com/MushroomObserver/mushroom-observer/assets/3211730/a349167b-8380-42a7-9ea6-6aa74fd45e68">

At first I thought it might be due to SolidQueue not running, but I started it up and I'm seeing the same problem.

JoeCohen commented:
I thought this was on the test server. Am I wrong?

On Fri, Jun 14, 2024 at 4:53 PM Nathan Wilson ***@***.***>
wrote:

> Hmm, I tried running it and the large images aren't showing up. The
> smaller images along the bottom are (when there's more than one of them).
> Screenshot.2024-06-14.at.7.49.03.PM.png (view on web)
> <https://github.com/MushroomObserver/mushroom-observer/assets/3211730/82511414-5674-4204-9567-4f84d9aa0adc>
> Screenshot.2024-06-14.at.7.47.56.PM.png (view on web)
> <https://github.com/MushroomObserver/mushroom-observer/assets/3211730/a349167b-8380-42a7-9ea6-6aa74fd45e68>
>
> At first I thought it might be due to SolidQueue not running, but I
> started it up and I'm seeing the same problem.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/MushroomObserver/mushroom-observer/pull/2155#issuecomment-2168931915>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAALDFEDCSRAGA3NBVONIMLZHN7ANAVCNFSM6AAAAABIQFB42KVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNRYHEZTCOJRGU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

mo-nathan commented:
Why do I have to run it on the test server?  That seems wrong.

mo-nathan commented:
test.mushroomobserver.org does work better.

Initial Comments:
1) I think the first field being "Notes" is kind of odd.  It's not immediately clear if this is notes about the image or the observation,  Given that observation notes can be more structured I think this needs to be clarified.  Assuming they are "Image Notes", I would label them as such and I'd put them at after the "Camera Info".
2) I think there should be an option for creating the observation above the fold.
3) Just adding the images and then clicking "Create" caused an observation to be created with no images.

JoeCohen commented:
@nathan: I don't think you have to run it on the test server. But for me,
it's easier than switching branches.

On the test server, type-ahead did not work (for either Location or
Scientific Name).

The lat/lng and date autofills work great.
Incomplete thinking out loud: Since those will be auto-filled in most
cases, would it make sense to put them after the fields which always
require manual input?

I agree with Nathan about clarifying the image Notes field.

@nathan does "option for creating the observation above the fold" mean you
want an additional Create button?
Not a bad idea. Maybe (at a later point) instead have a single floating
button if that's easy.


On Fri, Jun 14, 2024 at 6:39 PM Nathan Wilson ***@***.***>
wrote:

> test.mushroomobserver.org does work better.
>
> Initial Comments:
>
>    1. I think the first field being "Notes" is kind of odd. It's not
>    immediately clear if this is notes about the image or the observation,
>    Given that observation notes can be more structured I think this needs to
>    be clarified. Assuming they are "Image Notes", I would label them as such
>    and I'd put them at after the "Camera Info".
>    2. I think there should be an option for creating the observation
>    above the fold.
>    3. Just adding the images and then clicking "Create" caused an
>    observation to be created with no images.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/MushroomObserver/mushroom-observer/pull/2155#issuecomment-2169029319>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAALDFBKZOHQVKTNTM2UCLDZHOLOZAVCNFSM6AAAAABIQFB42KVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNRZGAZDSMZRHE>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

JoeCohen commented:
PS:
It created an Observation without images, even though the form had 2 images
(and the controller grabbed date and lat/lng from them).

On Fri, Jun 14, 2024 at 7:23 PM Joseph Cohen ***@***.***> wrote:

> @nathan: I don't think you have to run it on the test server. But for me,
> it's easier than switching branches.
>
> On the test server, type-ahead did not work (for either Location or
> Scientific Name).
>
> The lat/lng and date autofills work great.
> Incomplete thinking out loud: Since those will be auto-filled in most
> cases, would it make sense to put them after the fields which always
> require manual input?
>
> I agree with Nathan about clarifying the image Notes field.
>
> @nathan does "option for creating the observation above the fold" mean
> you want an additional Create button?
> Not a bad idea. Maybe (at a later point) instead have a single floating
> button if that's easy.
>
>
> On Fri, Jun 14, 2024 at 6:39 PM Nathan Wilson ***@***.***>
> wrote:
>
>> test.mushroomobserver.org does work better.
>>
>> Initial Comments:
>>
>>    1. I think the first field being "Notes" is kind of odd. It's not
>>    immediately clear if this is notes about the image or the observation,
>>    Given that observation notes can be more structured I think this needs to
>>    be clarified. Assuming they are "Image Notes", I would label them as such
>>    and I'd put them at after the "Camera Info".
>>    2. I think there should be an option for creating the observation
>>    above the fold.
>>    3. Just adding the images and then clicking "Create" caused an
>>    observation to be created with no images.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/MushroomObserver/mushroom-observer/pull/2155#issuecomment-2169029319>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAALDFBKZOHQVKTNTM2UCLDZHOLOZAVCNFSM6AAAAABIQFB42KVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNRZGAZDSMZRHE>
>> .
>> You are receiving this because you commented.Message ID:
>> ***@***.***>
>>
>

nimmolo commented:
@mo-nathan - The fact that the new form is letting you create an obs w/o a location sounds like a validation failure. I didn't even think to check it because i don't remember changing anything in the validation methods of the create action. Thank you.

The screengrabs from your local machine tell me your scss is not compiling. There may be a deeper problem with `dartsass-sprockets`, but it's more likely you have old assets lingering. 

You may know this, but as a rule of thumb: **We should only precompile assets on production.** Try to remember never to run `assets:precompile` locally, without running `assets:clobber` immediately afterward to remove the compiled assets, or you will get stuck in a CSS time warp, having to manually precompile every time you want an upstream change to be reflected locally.

The presence of any files in public/assets (from precompiling) will block live SCSS compilation on your machine indefinitely, because these folders are gitignored! Maybe also check for app/assets/builds. `assets:clobber` is your Tardis back to the present. 

And getting freshly-compiled CSS on dev will become increasingly relevant in the near future while we are updating all the CSS. 

@JoeCohen I haven't been able to replicate the obs being created with the images getting lost, but on test, i do get to the obs show page sometimes before the images have been transferred. This makes them appear blank at first. On reload, for me they are there.

nimmolo commented:
I should have pointed this out — note that observations _created on the test server_ are also _created in the production database_, so if you don't want to keep your experiments, you may want to delete them.

@JoeCohen if [this is the obs](https://test.mushroomobserver.org/550754) you tried to create with images, where the images were not attached to the obs, I think i found it! They sure aren't there, are they?

nimmolo commented:
@mo-nathan Observations with lat/lng but no Locality are currently permitted, and I think what you discovered is that this new form streamlines creating such observations, which is temporary but not ideal. Here's the logic in the controller:
```ruby
# app/controllers/observations_controller/new_and_create.rb
  def determine_observation_location(observation)
    if Location.is_unknown?(observation.place_name) ||
       (observation.lat && observation.lng && observation.place_name.blank?)
      observation.location = Location.unknown
      observation.where = nil
    end
    observation
  end
```
My idea for phase 2 was to make it so the form calls a new autocompleter that 
- suggests **only MO Locations matching your lat/lngs**
- reverse geolocates your lat/lng if we don't have a matching Location, and automatically creates it in the DB using whatever Google returns for a "smallest encompassing" place, preferring a `colloquial_area`, `locality` or `sublocality` and using its bounding box. This will cost us something for lookups, though.

Maybe what we need to do is to work this "best guess location" in to phase 1, though.

nimmolo committed:
Update new_and_create.rb

nimmolo committed:
Move and rename image notes field

nimmolo commented:
New consideration: if we want to order our Locality suggestions by increasing _area_, we should convert our location columns to geometric data so that MySQL can calculate the area for us, and orderby this value.

(If responding to this thread please do so within Github rather than via email if you can, because i've edited my responses above.)

mo-nathan commented:
In my view locality should not be required and in fact the current form claims it's required, but it's not and it defaults to "Earth".  This seems reasonable to me (but we should probably removed the "(required)" test.  Doing reverse geolocation would be awesome, but to me it is not required to get this phase of the work done.

mo-nathan commented:
In my view it is bad that test.mushroomobserver.org is writing to the production database.  Given that I will stop using it.  We should point it at it's own database.

JoeCohen commented:
I don't understand.
In particular, is a separate server worth the trouble and expense if it's
writing to a separate db?



On Sat, Jun 15, 2024 at 4:52 AM Nathan Wilson ***@***.***>
wrote:

> In my view it is bad that test.mushroomobserver.org is writing to the
> production database. Given that I will stop using it. We should point it at
> it's own database.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/MushroomObserver/mushroom-observer/pull/2155#issuecomment-2169388890>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAALDFHTO7VR4U7WNYLAICTZHQTI3AVCNFSM6AAAAABIQFB42KVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNRZGM4DQOBZGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

mo-nathan commented:
Hmm, I guess this is a matter of perspective.  It's even possible that I suggested this configuration, but now that I'm experiencing it, it seems wrong to me.  For me, a "test" server should be for testing so you can do crazy things with it to see if you can break something.  On a daily basis, I don't have content that I can just upload to production for testing so I don't know how I would do through testing like Nimmo asked for.  In my view, the right thing to do at the current stage of development is local testing.  However, that didn't work for me, so I got lured into thinking that "test.mushroomobserver.org" was an actual test server.  Maybe we should rename it to "preview" or just scrap the idea altogether.  At work we have a set of "lower" environments for various types of testing.  We start with local systems, then we have development (6), qa (2), loading testing (3), performance testing, staging, and user acceptance (4) servers that are all independent of actual production.  While we certainly don't need to go to this extreme, I think what we're doing at work is more what the expected model is.  The idea of running separate servers that point to the same database is inherently problematic since bugs in one of them may mess with the other one in pretty significant ways.

JoeCohen commented:
@nimmo: That's the problematic obs.
>@JoeCohen if [this is the obs](https://test.mushroomobserver.org/550754) you tried to create with images, where your images were not attached to the obs, I think i found it. They sure aren't there, are they? 😕

nimmolo commented:
@mo-nathan Did `assets:clobber` fix your dev environment?

nimmolo commented:
@mo-nathan — I agree that "test" implies that you are encouraged to break stuff. But for now, what say we try to get through these obs-form PRs by continuing to test on "test" and just deleting our obs afterwards. That's what I've been doing — I've created untold "Corallorhiza" observations and deleted them. 

It was a lot of work for me setting up the server on `test` again, because of my unfamiliarity setting up Linux servers — and just a couple months after giving the go-ahead to delete the previous test server. I do feel that we need one for the type of PRs we have in the pipeline currently.

nimmolo commented:
@JoeCohen Are you able to replicate the disappearing images issue? Or suggest anything I can do to try to get the same result?

JoeCohen commented:
@nimmolo: I cannot replicate that result. (I just tried several times.)

nimmolo committed:
relevant clean ups

Fix listener syntax (create-obs-reorg)
Eliminate commented out template in filter

nimmolo committed:
create-obs-reorg fix action on item

nimmolo committed:
Update _where.html.erb

Remove `required`

nimmolo committed:
Temporarily re-add another create button

Form should be tabbed soon

nimmolo commented:
Revisions deployed to `test`. Please re-review

- Re-added the extra "Create" button. (Buttons will all likely be very different with the tabbed form in part 3 of these PRs)
- Renamed "Image Notes" - I agree this disambiguation is better. 
- Also moved it down below camera info. I'm not crazy about this, though. Having it at the bottom of the form seems to discourage people from writing captions, which I'd rather people write. Anyway, we can all consider this placement. 
- I'm not able to reproduce the "obs created with no images" bug. But please try to!

mo-nathan commented:
Did another create and got a broken image after just clicking create.  However, the image showed up after a reload.  Note that I was trying to get the observation to be created with as little info as possible.  I just dropped the image and then scrolled down and clicked "Create".

mo-nathan commented:
I also noticed that if you use the default location of "Earth" that clicking on the lat/long doesn't give you any thing useful.  I was hoping to see the lat/long on a map so I could remember exactly where that photo was from.

nimmolo committed:
Move "image notes" back up

nimmolo commented:
@mo-nathan - I believe the broken image is because the test server is slow.

I'm saving work on coordinating the Location, lat/lngs and the two(!!) maps til the next PR. 

Note that there are two maps in this form — this is not new, although I find it confusing. There's a "Open interactive map" for the lat/lng, and there's a "Locate on map" for box locations. Using "Earth" as a location for the latter will disable the second map entirely, as I believe it should.

In the tabbed form of the mobile app, the lat/lng  in the first tab with the photo because it is derived from it — i.e., in a separate tab from the "Locality". I'm aiming to model our tabbed form after the mobile app, but let's discuss it.

So my current thinking is that the tabbed form may keep these two map UIs, so people can check their map lat/lng and their locality in each tab. I feel like the "locality" map should probably show the lat/lng point if there is one, though, so people can catch the correspondence to the locality — especially because we're going to be suggesting and in some cases auto-filling from an MO location, or auto-creating a Location from Google data.

mo-nathan commented:
@nimmolo It makes me nervous that the test server being slow is causing a broken image.  I'm thinking of the case where there are bunch of folks uploading images to MO at an event. I really want to test this against production, but it seems like something more graceful than a broken image would still make sense.

Regarding the map, I can't tell whether you think a lat/long listing with no location (== "Earth") is working as it should or if you're just saying that it's a separate issue (which I can agree with).  The current behavior doesn't make sense to me.  Why should it behave differently if I have "Earth" vs. "USA", but provide a precise lat/long (as it does at the moment).

nimmolo commented:
@mo-nathan - I understand why that would make you nervous. 

It occurred to me to check the behavior of the current `main` branch if it's running on the `test` server — do images appear with dead urls at first, and then they appear on reload? I just tried it, and that is the case. I've still got `main` running on `test`, so let me know if you get the same results. It may not be because the server is slow — it may be a config that I got wrong when setting up the server in the first place. I've sent a question to Jason about it.

Re maps: separate issue, yes. But I think where the current UI is wrong is that once you've opened the map, there's no longer a button to mark your lat/lng on the map! Our map JS hides that button once the map is open. That does not seem right. You can only map the "Location" (box) you're picking once the map is open, you can't re-center it on your lat/lng (point).

mo-nathan commented:
Re: the test server behavior, that makes some sense to me and I'll look at it this evening.

Re: the map behavior, I'm not sure I've explained the problem clearly enough.  I'll create an issue this evening with screen shots.  Hopefully I can find some time to fix it over the weekend.  I've been working on understanding the gap between how field slips and project constraints should work and how they currently work on the site so I can get that sorted.

nimmolo commented:
@mo-nathan - Jason figured out that it was indeed a config, and fixed it. We're still ironing out the configs to make sure this doesn't happen again, and to make the code portable to other hypothetical test servers. I'll commit whatever changes there are that should go into the repo.

I'm going to switch the test server back to this branch, and test it again. I'll also ask you and @JoeCohen to kick the tires one more time, and merge tonight if that's ok. I'm pretty far along on the second "location suggestions" PR.

nimmolo commented:
@mo-nathan re: map behavior - yes, thank you. Rereading this thread i'm not sure I understood very well either, and in fact I know i described some things inaccurately a couple responses above. I've tried to correct those by editing the responses. But an issue with screenshots will make things clearer, i imagine.

mo-nathan reviewed (changes requested):
I do not think we should push these changes out until this works in a local environment.  Right now when I try to run it and load an image does not show me the image.  It does appear to pick up on the EXIF data, but no image is shown.  If you add a second image it shows both images in the thumbnails at the bottom, but nothing I do causes the full image to show in the carousel.
<img width="1210" alt="Screenshot 2024-06-20 at 7 20 30 PM" src="https://github.com/MushroomObserver/mushroom-observer/assets/3211730/54bf1b8c-975f-48fb-8cb7-bc471e7590e9">
<img width="518" alt="Screenshot 2024-06-20 at 7 21 02 PM" src="https://github.com/MushroomObserver/mushroom-observer/assets/3211730/92f5db97-573f-4d40-9fb8-68077dabf482">.

mo-nathan commented:
In my view this rubocop violation should be addressed:
TODO found NEW
# TODO: make this a component
Severity: MinorFound in [app/views/controllers/observations/form/images/_carousel.erb](https://codeclimate.com/github/MushroomObserver/mushroom-observer/app/views/controllers/observations/form/images/_carousel.erb?from_sha=b42b7a74&to_sha=c70f5e3d) by [fixme](https://docs.codeclimate.com/docs/fixme)

mo-nathan commented:
If you want me to "kick the tires", I need to be able to run it on my local so I freely test it without impacting the production database.

mo-nathan commented:
At the moment at least test.mushroomobserver.org doesn't have the new create observation UI.
<img width="1434" alt="Screenshot 2024-06-20 at 7 29 30 PM" src="https://github.com/MushroomObserver/mushroom-observer/assets/3211730/f7888d6f-b8cc-4db8-b612-1e431d18406a">

mo-nathan commented:
I looked at test.mushroomobserver.org because I noticed that locally this branch had the "Image notes" at the top of the panel.  This still doesn't feel like the right thing to be expecting users to be doing first.  I expect the majority of images to not have notes.  I'm not saying we should do exactly the same thing but if you look at the workflow for iNat, they streamline it down to providing the name and inferring the date and location.  The last thing they ask for are notes which are presumably notes for the observation not for the individual image.

I also think that "When it was taken" should be above the "Copyright holder" and the "License" selector should be right next to the "Copyright holder".

mo-nathan commented:
Went back to try it locally and noticed that the text in the "Camera info" box looks off.  Specifically it says "Use this infoInfo copied".  I think it's intended to just say "Info copied" since the button is disabled, but I'm not 100% sure.

mo-nathan commented:
When there is a second image uploaded the button becomes active, but it still says "Use this infoInfo copied".

mo-nathan commented:
Given that the plan is for the location and date info to be inferred from the photos and we expect that to be used a lot, I find myself agreeing with iNat that the name selection should be higher on the page.

nimmolo commented:
Responding on Slack!

mo-nathan commented:
Documented the map issue here: https://github.com/MushroomObserver/mushroom-observer/issues/2187

nimmolo committed:
Merge branch 'main' into create-obs-reorg

nimmolo committed:
Merge branch 'main' into create-obs-reorg

nimmolo committed:
Fix handling of image exif transfer button

Don't offer to transfer if the data's not there

nimmolo committed:
Hide use exif button, don't gray out, if no data

nimmolo committed:
Change map button names and behavior

Keep map button to "Find point on map" so that people can correlate the lat/lng of an attached image with a map point, to double check it.

JoeCohen commented:
Problems with maps while creating Observations on the test server. (Sorry if I've overlooked something the conversations here, on email, or Slack.)
- Create Observation
- Drag photo to page
Results: 
👍 Date fills automatically
👍 lat/lng fills automatically
- Click `Open Interactive Map` button
Result:
👎 `Open Interactive Map` button disappears, but there's no map.

 - Create Observation
- Drag photo to page
- Click `Hide exact coordinates?` checkbox
- Click `Open Interactive Map` button

JoeCohen commented:
Suggestions re form:
1. More prominent instructions about Selecting/Dropping Files. Also maybe include something like "Step One".  
2. Align geoloc help with Lat/Lon/Elevation (rather than Open Interactive Map).
<img width="1205" alt="Screenshot 2024-06-22 at 1 03 12 PM" src="https://github.com/MushroomObserver/mushroom-observer/assets/45460/69f62eb8-1998-4d23-9b7c-36934f2a9dea">

nimmolo committed:
Merge branch 'main' into create-obs-reorg

nimmolo committed:
Merge branch 'main' into create-obs-reorg

nimmolo commented:
@JoeCohen - I changed the button from "Open interactive map" to "Find point on map", and changed the behavior so that the button is not hidden when the map opens.

However! The Google Maps API will not serve the map to `test.mushroomobserver.org` because it's not a permitted domain for our API key. If you know how to permit it, that could be helpful for our testers too.

nimmolo commented:
I think I just permitted test.mushroomobserver.org. Let's check if it works...

JoeCohen commented:
You did permit it, and it fixes those issues.

nimmolo committed:
Merge branch 'main' into create-obs-reorg

nimmolo committed:
Merge branch 'main' into create-obs-reorg

nimmolo committed:
params[:good_image_ids]

nimmolo committed:
Merge branch 'main' into create-obs-reorg

nimmolo committed:
Merge branch 'main' into create-obs-reorg

nimmolo committed:
CSS - carousel controls and overall

mo-nathan commented:
This is looking pretty good.  What's left on your list?  I spotted a bit of cleanup that you mentioned last night (like getting rid of the ID text box for locations and names and getting the "When" fields to be arranged horizontally rather than vertically when in mobile mode).

mo-nathan commented:
I wonder if we should have a warning popup if you try to navigate away from the create observation page after you've uploaded an image.  Given the complexity of that UI at this point I kind of lost track that it wasn't permanent when I went to change the theme to test that out.  I don't think it would harm anything to just have a "Are you sure you don't want to create this observation?" modal popup if they have put in enough effort to have uploaded an image.

mo-nathan commented:
Hmm, it's a bit more complicated than that.  I see now that if you go to a read only page (like click on the "License" link) and then go back to the Create Observation page, then the state is saved.  Even if you go to the get page for updating your preferences and immediately go back, the state is saved.  However, if you change the state in preferences and do a save, then when you go back the state for the Create Observation page is lost.  Oh and if you go to the "License" link and then click "Create Observation" rather than using the back button, then it initially shows the previous state which then vanishes and you have a blank form.  This is definitely a tricky case.  It make me think that these links should create new tabs, but that's a pretty big change and at least used to be frowned on by the HTML standards folks.  However, it seems a lot more common these days.  No idea if there's an easy way to do that generally from the Create Observation page and not from pages that don't have any relevant state.

nimmolo committed:
Re-hide the ID field

nimmolo commented:
We could execute some JS that adds a `target="_blank"` attribute to all links on the Create Obs page.

On the other hand, that adds complexity and i don't think users will make that mistake twice - they can always right-click a link to open in a new tab.

nimmolo committed:
Update _carousel.erb

nimmolo commented:
@JoeCohen - I moved the geo help up alongside lat/lng/alt as you suggested. 
I hope to redesign these help boxes in the next phase PR, so it may not be there long.

nimmolo committed:
Obs date selects and lat/lng/alt - inline even on mobile

nimmolo commented:
@mo-nathan Removed the `TODO` from the carousel template

nimmolo commented:
Merging phase 1.
